src/SpinWaveTheory/SWTCalculations.jl

```diff
@@ -15,20 +18,14 @@ function swt_hamiltonian!(swt::SpinWaveTheory, k̃ :: Vector{Float64}, Hmat::Mat
     (; sys, s̃_mat, T̃_mat, Q̃_mat) = swt
     Hmat .= 0 # DD: must be zeroed out!
     Nm, Ns = length(sys.dipoles), sys.Ns[1] # number of magnetic atoms and dimension of Hilbert space
+    # Below two lines are for dipole mode only
+    S = (Ns-1) / 2 # spin-S
+    biquad_res_factor = 1 - 1/S + 1/(4S^2) # rescaling factor for biquadratic interaction
+
     Nf = sys.mode == :SUN ? Ns-1 : 1
     N  = Nf + 1
     L  = Nf * Nm
     @assert size(Hmat) == (2*L, 2*L)
-    # scaling factor (=1) if in the fundamental representation
-    M = sys.mode == :SUN ? 1 : (Ns-1)
```

Comment:
Sorry if I missed this. Why to the `M` scalings go away in your revision?

Reply:
See above